microprofile/health-smallrye/src/main/java/org/wildfly/extension/microprofile/health/deployment/CDIExtension.java

```diff
@@ -51,84 +45,66 @@
 
     private final HealthReporter reporter;
 
-    private List<AnnotatedType<? extends HealthCheck>> healthDelegates = new ArrayList<>();
-    private Collection<Unmanaged.UnmanagedInstance<HealthCheck>> healthInstances = new ArrayList<>();
+    static final class HealthLiteral extends AnnotationLiteral<Health> implements Health {
 
-    private List<AnnotatedType<? extends HealthCheck>> livenessDelegates = new ArrayList<>();
-    private Collection<Unmanaged.UnmanagedInstance<HealthCheck>> livenessInstances = new ArrayList<>();
+        static final HealthLiteral INSTANCE = new HealthLiteral();
 
-    private List<AnnotatedType<? extends HealthCheck>> readinessDelegates = new ArrayList<>();
-    private Collection<Unmanaged.UnmanagedInstance<HealthCheck>> readinessInstances = new ArrayList<>();
+        private static final long serialVersionUID = 1L;
 
+    }
+    static final class LivenessLiteral extends AnnotationLiteral<Liveness> implements Liveness {
```

Comment:
Would it be worth MP Health spec issue to add these literals there in the future?
Ideally, along with `@Target` on those qualifiers ;)